web3-react/src/libs/connections.ts

```diff
@@ -0,0 +1,126 @@
+import { Web3ReactHooks } from '@web3-react/core'
+import { Connector, AddEthereumChainParameter } from '@web3-react/types'
+import { buildInjectedConnector } from './injected'
+import { buildNetworkConnector } from './network'
+import { buildCoinbaseWalletConnector } from './coinbase'
+import { buildGnosisSafeConnector } from './gnosis'
+import { buildWalletConnectConnector } from './wallet-connect'
+import { CHAIN_INFO } from './constants'
+
+export interface Connection {
+  connector: Connector
+  hooks: Web3ReactHooks
+  type: ConnectionType
+}
+
+export enum ConnectionType {
+  COINBASE_WALLET = 'COINBASE_WALLET',
+  GNOSIS_SAFE = 'GNOSIS_SAFE',
+  INJECTED = 'INJECTED',
+  NETWORK = 'NETWORK',
+  WALLET_CONNECT = 'WALLET_CONNECT',
+}
+
+export function getIsInjected(): boolean {
+  return Boolean(window.ethereum)
+}
+
+export function getHasMetaMaskExtensionInstalled(): boolean {
+  return window.ethereum?.isMetaMask ?? false
+}
+
+export function onConnectionError(error: Error) {
+  console.debug(`web3-react error: ${error}`)
+}
+
+export const PRIORITIZED_CONNECTORS: Connection[] = [
+  buildInjectedConnector(),
+  buildCoinbaseWalletConnector(),
+  buildWalletConnectConnector(),
+  buildGnosisSafeConnector(),
+  buildNetworkConnector(),
+]
+
+export function getConnection(c: Connector | ConnectionType) {
+  if (c instanceof Connector) {
+    const connection = PRIORITIZED_CONNECTORS.find(
+      (connection) => connection.connector === c
+    )
+    if (!connection) {
+      throw Error('unsupported connector')
+    }
+    return connection
+  } else {
+    switch (c) {
+      case ConnectionType.INJECTED:
+        return PRIORITIZED_CONNECTORS[0]
```

Comment:
What about using a map?
```ts
const map = { [ConnectionType.INJECTED]: buildInjectedConnector() }
```
and then, calling it `map[ConnectionType.INJECTED]` instead?